user-ssh-builders/README.md

```diff
@@ -0,0 +1,142 @@
+# Give users SSH access to builders
+
+Author: [@FrostyX](https://github.com/frostyx)
+
+
+## Motivation
+
+Users sometimes need SSH access to a builder instance to debug Copr-only
+build failures. We want them to be able to get the access without
+admin intervention.
+
+See also https://github.com/fedora-copr/copr/issues/2364
+
+
+## Limits
+
+Obviously, Copr resources aren't unlimited so we need to allocate them
+wisely.
+
+- Only 1 running instance per user
+    - Having 1 instance per project would be too much
+    - Having more than 1 instance per user isn't needed, users can
+      team up for debugging issues that need more builders at once
+    - No need for group instances, users can do whatever they want,
+      including adding access to their team members
+
+- Builders need to have a limited lifespan, e.g. 48 hours
+    - When submitting a build in Copr, we allow to adjust the timeout
+      to 30 hours at most. That means the lifespan needs to be at
+      least 30 hours.
+    - I propose 48 hours because we want to give users enough time for
+      debugging after the build finishes
```

Comment:
I would give 1/2 or 1 hour max by default, with the possibility to prolong to 2 days as you propose.

Reply:
+1 for having a really limited lifespan by default. Most of the people won't need much. Half an hour looks fine to me. Not sure if you have a median of the build time or some other metric we can compare this to.